src/main/java/com/example/demo/services/UserService.java

```diff
@@ -0,0 +1,82 @@
+package com.example.demo.services;
+
+import com.example.demo.dtos.ProductDTO;
+import com.example.demo.entities.User;
+import com.example.demo.mappers.DtoMapper;
+import com.example.demo.products.Product;
+import com.example.demo.repositories.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+@Service
+@RequiredArgsConstructor
+public class UserService {
+
+    @Autowired
+    public static UserRepository userRepository;
+    private final DtoMapper mapper;
+
+    public Page<User> pageUsers(Pageable pageRequest){
+        return userRepository.findAll(pageRequest);
+    }
+
+    public void banUser(Long id) {
+        userRepository.findById(id).ifPresent(User -> {
+            User.setActive(false);
+            userRepository.save(User);
+        });
+    }
+
+    public void addProduct(User user, Product product) {
+        user.getProducts().add(product);
+        userRepository.save(user);
+    }
+
+    public void deleteProduct(User user, int index) {
+        user.getProducts().remove(index);
+        userRepository.save(user);
+    }
+
+    public void updateProduct(User user, int index, ProductDTO product) {
+        List<Product> products = user.getProducts();
+        String productType = products.get(index).getClass().getName();
+        switch (productType){
```

Comment:
service should not be aware of any DTOs due to law of Demeter

Reply:
Updated (not sure if it's right way to do it but idk how else to do it)